invenio_drafts_resources/services/records/service.py

```diff
@@ -263,6 +266,21 @@ def update_draft(
         """Replace a draft."""
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
 
+        # can not make record restricted after grace period
+        if (
+            not current_app.config["ALLOW_RECORD_RESTRICTION_AFTER_GRACE_PERIOD"]
+            and data.get("access", {}).get("record", None) == "restricted"
+            and draft.get("access", {}).get("record", None) == "public"
+        ):
+            end_of_grace_period = (
+                draft.created + current_app.config["RECORDS_RESTRICTION_GRACE_PERIOD"]
```

Comment:
we assume here that grace period is configured, do we want to provide the fallback in case it is not, or to fail? I think the failure is a bit too much user exposed, because it will be only discovered during runtime when draft is being updated. To be discussed.